src/providers/goDaddy.ts

```diff
@@ -58,14 +58,24 @@ export const setRecord = async (
       type: 'A'
     },
     {
-      data: domain,
-      name: domain,
+      data: '@',
+      name: '*',
       type: 'CNAME'
+    },
+    {
+      data: 'ns01.domaincontrol.com',
+      name: '',
+      type: 'NS'
+    },
+    {
+      data: 'ns02.domaincontrol.com',
+      name: '',
+      type: 'NS'
     }
   ]
 
   const options = {
-    method: 'PATCH',
+    method: 'PUT',
```

Comment:
I don't think `PUT` is a good idea, this would get rid of existing `MX` records and stuff.
Steps:
1. Get existing A and CNAME Records.
2. Patch / Delete problematic records.
`A` record should have only one `@` entry.
`CNAME` record must have only one `*` entry, pointing to `@`.